pgmq-extension/sql/pgmq.sql

```diff
@@ -378,18 +378,24 @@ END
 $$ LANGUAGE plpgsql;
 
 -- purge queue, deleting all entries in it.
-CREATE FUNCTION pgmq."purge_queue"(queue_name TEXT)
+CREATE OR REPLACE FUNCTION pgmq."purge_queue"(queue_name TEXT)
 RETURNS BIGINT AS $$
 DECLARE
   deleted_count INTEGER;
   qtable TEXT := pgmq.format_table_name(queue_name, 'q');
 BEGIN
-  EXECUTE format('DELETE FROM pgmq.%I', qtable);
-  GET DIAGNOSTICS deleted_count = ROW_COUNT;
+  -- Get the row count before truncating
+  EXECUTE format('SELECT count(*) FROM pgmq.%I', qtable) INTO deleted_count;
```

Comment:
```suggestion
  EXECUTE format('SELECT count(*) FROM pgmq.%I', qtable) INTO deleted_count;
```

Considering the goal of this function is performance, I think it's fair to change our description to "estimation of number of rows deleted", and use the reltuples stat. I don't think the exact number of messages deleted is too important anyway.

Reply:
Just an idea, though.

Reply:
then  shall i modify it ??

Reply:
Let's leave it for now since current behavior tells user how many are deleted. We can consider dropping it for the v2.0 release of the project though.